app/src/main/java/com/example/concordia_campus_guide/Activities/MainActivity.java

```diff
@@ -92,35 +88,51 @@ public boolean onOptionsItemSelected(MenuItem item) {
      */
     public void showInfoCard(String buildingCode){
         if(infoCardFragment!=null){
-            hideInfoCard();
+            resetBottomCard();
         }
 
         infoCardFragment = new InfoCardFragment(buildingCode);
         fragmentTransaction = fragmentManager.beginTransaction();
-        fragmentTransaction.add(R.id.info_card_frame, infoCardFragment);
+        fragmentTransaction.remove(poiFragment);
+        fragmentTransaction.add(R.id.bottom_card_frame, infoCardFragment);
         fragmentTransaction.commit();
 
         swipeableInfoCard.setState(BottomSheetBehavior.STATE_COLLAPSED);
 
     }
 
-    /**
-     * Hides the info card fragment from the view.
-     */
-    public void hideInfoCard(){
+    public void showPOICard(){
+        poiFragment = new POIFragment();
         fragmentTransaction = fragmentManager.beginTransaction();
-        fragmentTransaction.remove(infoCardFragment);
+        fragmentTransaction.add(R.id.bottom_card_frame, poiFragment);
         fragmentTransaction.commit();
+
+        swipeableInfoCard.setState(BottomSheetBehavior.STATE_COLLAPSED);
+        swipeableInfoCard.setPeekHeight(180);
+        swipeableInfoCard.setHideable(false);
+
+    }
+
+    public void resetBottomCard(){
```

Comment:
I'd suggest changing the content of this method to only remove fragments in the bottom card (as per our discussion) instead of removing all fragments that are not LocationFragment. This can be done by creating a list that can contain all the fragments that will be put in the bottom card or a list that contains the frangment classes that are added to bottom cards.